src/Okta.Sdk.UnitTests/ApplicationsClientShould.cs

```diff
@@ -130,6 +131,61 @@ public async Task BuildRevokeTokensRequest()
             mockRequestExecutor.ReceivedHref.Should().StartWith("/api/v1/apps/foo/tokens");
         }
 
+        [Fact]
+        public async Task BuildPublishCerCertRequest()
+        {
+            var mockRequestExecutor = new MockedStringRequestExecutor(string.Empty);
+            var client = new TestableOktaClient(mockRequestExecutor);
+
+            await client.Applications.PublishCerCertAsync("fakeBase64EncodedCertificate", "testAppId", "testCertificateSigningRequestId");
+
+            mockRequestExecutor.ReceivedHref.Should().Be("/api/v1/apps/bar/credentials/csrs/baz/lifecycle/publish");
+        }
+
+        [Fact]
+        public async Task BuildPublishBinaryCerCertRequest()
+        {
+            var mockRequestExecutor = new MockedStringRequestExecutor(string.Empty);
+            var client = new TestableOktaClient(mockRequestExecutor);
+
+            await client.Applications.PublishBinaryCerCertAsync(Encoding.UTF8.GetBytes("fakeBase64EncodedCertificate"), "testAppId", "testCertificateSigningRequestId");
+
+            mockRequestExecutor.ReceivedHref.Should().Be("/api/v1/apps/bar/credentials/csrs/baz/lifecycle/publish");
+        }
+
+        [Fact]
+        public async Task BuildPublishDerCertRequest()
+        {
+            var mockRequestExecutor = new MockedStringRequestExecutor(string.Empty);
+            var client = new TestableOktaClient(mockRequestExecutor);
+
+            await client.Applications.PublishDerCertAsync("fakeDerCertificate", "testAppId", "testCertificateSigningRequestId");
+
+            mockRequestExecutor.ReceivedHref.Should().Be("/api/v1/apps/bar/credentials/csrs/baz/lifecycle/publish");
+        }
+
+        [Fact]
+        public async Task BuildPublishBinaryDerCertRequest()
+        {
+            var mockRequestExecutor = new MockedStringRequestExecutor(string.Empty);
+            var client = new TestableOktaClient(mockRequestExecutor);
+
+            await client.Applications.PublishBinaryDerCertAsync(Encoding.UTF8.GetBytes("foo"), "bar", "baz");
+
+            mockRequestExecutor.ReceivedHref.Should().Be("/api/v1/apps/bar/credentials/csrs/baz/lifecycle/publish");
+        }
+
+        [Fact]
+        public async Task BuildPublishBinaryPemCertRequest()
+        {
+            var mockRequestExecutor = new MockedStringRequestExecutor(string.Empty);
```

Comment:
For PublishCsr* methods, let's verify that headers and encoding are being sent correctly.

Reply:
Per our conversation, this won't be possible here due to the nature of the underlying implementation.  Validation for the actual headers occurs in other tests.